osu.Game/Scoring/ScoreInfo.cs

```diff
@@ -183,6 +183,9 @@ protected class DeserializedMod : IMod
 
         public override string ToString() => $"{User} playing {Beatmap}";
 
-        public bool Equals(ScoreInfo other) => other?.OnlineScoreID == OnlineScoreID;
+        public bool Equals(ScoreInfo other) =>
+            other?.OnlineScoreID == OnlineScoreID
```

Comment:
just for readability and safety, i'd prefer doing a null check as the first condition here. right now the first condition (`other?.OnlineScoreID == OnlineScoreID`) can pass even if other is null (as `OnlineScoreID` is nullable)